bootstrap_cfn/fab_tasks.py

```diff
@@ -193,6 +195,112 @@ def swap_tags(tag1, tag2):
     r53_conn.update_dns_record(zone_id, fqdn2, 'TXT', '"{0}"'.format(stack_suffix1))
 
 
+@task
+def enter_maintenance(maintenance_ip):
+    """
+    Puts stack into maintenance mode
+
+    Sets all internet facing elb hostnames to resolve to given maintenance_ip
+    """
+    cfn_config = get_config()
+    r53_conn = get_connection(R53)
+
+    cached_zone_ids = {}
+    for elb in cfn_config.data['elb']:
+        if not elb['scheme'] == 'internet-facing':
+            continue
+
+        record = "{name}.{hosted_zone}".format(**elb)
+        zone_id = get_cached_zone_id(r53_conn, cached_zone_ids, elb['hosted_zone'])
+        print green("Attempting to update: \"{0}\":\"{1}\"".format(record, maintenance_ip))
+        r53_conn.update_dns_record(zone_id, record, 'A', maintenance_ip)
+
+
+@task
+def exit_maintenance():
+    """
+    Exit maintenance mode
+
+    Sets internet-facing elbs hostnames
+    back to the ELB DNS alias
+    """
+    r53_conn = get_connection(R53)
+    elb_conn = get_connection(ELB)
+
+    cfn_config = get_config()
+    stack_name = get_stack_name()
+
+    # In order to traverse from config yaml all the way to the DNS alias for the ELB
+    # it is required to construct a logical to physical naming for the elbs. So first
+    # get all elbs for this stack from AWS cloudformation, to be used as a
+    # filter on the next step
+    # Note: if stack does not exist this will throw a BotoServerError
+    stack_elbs = dict([
+        (x.logical_resource_id, x.physical_resource_id)
+        for x in elb_conn.cfn.get_stack_load_balancers(stack_name)])
+
+    # filter stack related load balancers (as opposed to all stack elbs in the account)
+    full_load_balancers = elb_conn.conn_elb.get_all_load_balancers(
+        load_balancer_names=stack_elbs.values())
+
+    cached_zone_ids = {}
+    # loop through elb config entries and change internet facing ones
+    for elb in cfn_config.data['elb']:
+        if not elb['scheme'] == 'internet-facing':
+            continue
```

Comment:
Since this loop and filter code is repeated (also on lines 209-211), it should probably be refactored into a function, eg:
```python
def internet_facing_elbs(cfn_config):
    return filter(lambda elb: elb['scheme'] == 'internet-facing', cfn_config.data['elb'])
```
or even better:
```python
def is_internet_facing(elb):
    return elb['scheme'] == 'internet-facing'

def internet_facing_elbs(cfn_config):
    return filter(is_internet_facing, cfn_config.data['elb'])

for elb in internet_facing_elbs(cfn_config):
    ...
```